libraries/ESP_NOW/src/ESP32_NOW.cpp

```diff
@@ -0,0 +1,387 @@
+    } else if (result == ESP_ERR_ESPNOW_NOT_FOUND) {
+        log_e("Peer Not Found");
+    }
+
+    for(uint8_t i=0; i<ESP_NOW_MAX_TOTAL_PEER_NUM; i++){
+        if(_esp_now_peers[i] != NULL && memcmp(mac_addr, _esp_now_peers[i]->addr(), ESP_NOW_ETH_ALEN) == 0){
+            _esp_now_peers[i] = NULL;
+            break;
+        }
+    }
+    return result;
+}
+
+static esp_err_t _esp_now_modify_peer(const uint8_t *mac_addr, uint8_t channel, wifi_interface_t iface, const uint8_t *lmk){
+    log_v(MACSTR, MAC2STR(mac_addr));
+    if (!esp_now_is_peer_exist(mac_addr)) {
+        log_e("Peer not found");
+        return ESP_ERR_ESPNOW_NOT_FOUND;
+    }
+
+    esp_now_peer_info_t peer;
+    memset(&peer, 0, sizeof(esp_now_peer_info_t));
+    memcpy(peer.peer_addr, mac_addr, ESP_NOW_ETH_ALEN);
+    peer.channel = channel;
+    peer.ifidx = iface;
+    peer.encrypt = lmk != NULL;
+    if(lmk){
+        memcpy(peer.lmk, lmk, ESP_NOW_KEY_LEN);
+    }
+
+    esp_err_t result = esp_now_mod_peer(&peer);
+    if (result == ESP_OK) {
+        return result;
+    } else if (result == ESP_ERR_ESPNOW_NOT_INIT) {
+        log_e("ESPNOW Not Init");
+    } else if (result == ESP_ERR_ESPNOW_ARG) {
+        log_e("Invalid Argument");
+    } else if (result == ESP_ERR_ESPNOW_FULL) {
+        log_e("Peer list full");
+    } else if (result == ESP_ERR_ESPNOW_NO_MEM) {
+        log_e("Out of memory");
+    }
+    return result;
+}
+
+static void _esp_now_rx_cb(const esp_now_recv_info_t *info, const uint8_t *data, int len){
+    log_v(MACSTR", data lenght : %d", MAC2STR(info->src_addr), len);
+    log_buf_v(data, len);
+    if(!esp_now_is_peer_exist(info->src_addr) && new_cb != NULL){
+        log_v("Calling new_cb, peer not found.");
+        new_cb(info, data, len, new_arg);
+        return;
+    }
+    //find the peer and call it's callback
+    for(uint8_t i=0; i<ESP_NOW_MAX_TOTAL_PEER_NUM; i++){
+        if(_esp_now_peers[i] != NULL && memcmp(info->src_addr, _esp_now_peers[i]->addr(), ESP_NOW_ETH_ALEN) == 0){
+            _esp_now_peers[i]->_onReceive(data, len);
+            return;
+        }
+    }
+}
+
+static void _esp_now_tx_cb(const uint8_t *mac_addr, esp_now_send_status_t status) {
+    log_v(MACSTR" : %s", MAC2STR(mac_addr), (status == ESP_NOW_SEND_SUCCESS)?"SUCCESS":"FAILED");
+    //find the peer and call it's callback
+    for(uint8_t i=0; i<ESP_NOW_MAX_TOTAL_PEER_NUM; i++){
+        if(_esp_now_peers[i] != NULL && memcmp(mac_addr, _esp_now_peers[i]->addr(), ESP_NOW_ETH_ALEN) == 0){
+            _esp_now_peers[i]->_onSent(status == ESP_NOW_SEND_SUCCESS);
+            return;
+        }
+    }
+}
+
+ESP_NOW_Class::ESP_NOW_Class(){}
+
+ESP_NOW_Class::~ESP_NOW_Class(){}
+
+bool ESP_NOW_Class::begin(const uint8_t *pmk){
+    if(_esp_now_has_begun){
+        return true;
+    }
+    _esp_now_has_begun = true;
+
+    memset(_esp_now_peers, 0, sizeof(ESP_NOW_Peer *) * ESP_NOW_MAX_TOTAL_PEER_NUM);
+
+    esp_err_t err = esp_now_init();
```

Comment:
I would add some kind of check to see if the WiFi was initialized. This will cause a LoadException if the WiFi is not initialized.

Reply:
Agree, will take a look.

Reply:
@lucasssvaz done